### PR TITLE
Added lint enum_struct_variant_field_now_doc_hidden

### PR DESCRIPTION
**Pull request by arpity22**
Lint for when a field in struct variant in public enum is now marked as #[doc(hidden)]

Issue mentioned in https://github.com/obi1kenobi/cargo-semver-checks/issues/578

Caught a lint in test crate for the lint enum_now_doc_hidden, which is mentioned in the test output for this lint

### CODE REVIEW

File: src/lints/enum_struct_variant_field_now_doc_hidden.ron [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+                    }
+                }
+            }
+        }
+    }"#,
+    arguments: {
+        "public": "public",
+        "true": true,
+    },
+    error_message: "A field of a pub enum tuple variant is now marked #[doc(hidden)] and is no longer part of the public API.",
```
Comment (obi1kenobi) [inline src/lints/enum_struct_variant_field_now_doc_hidden.ron:69]:
```suggestion
    error_message: "A field of a pub enum struct variant is now marked #[doc(hidden)] and is no longer part of the public API.",
```

**Review by obi1kenobi — APPROVED** [review]

**Comment by obi1kenobi** [comment]
Very nice! Marked for merge 🚀

If you're open to shipping more lints, I'm very happy to keep merging them!

**Comment by arpity22** [comment]
Maybe tomorrow I kinda have to study for my tests now😅

**Comment by obi1kenobi** [comment]
Of course, good luck with your tests!

**Pull request merged**